lib/internal/Magento/Framework/Config/View/DesignResolver.php

```diff
@@ -0,0 +1,195 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\Framework\Config\View;
+
+use Magento\Framework\App\Area;
+use Magento\Framework\App\Filesystem\DirectoryList;
+use Magento\Framework\Config\DesignResolverInterface;
+use Magento\Framework\Config\FileIteratorFactory;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\Filesystem;
+use Magento\Framework\Module\Dir\Reader as ModuleReader;
+use Magento\Framework\Phrase;
+use Magento\Framework\View\ConfigInterface;
+use Magento\Framework\View\Design\Fallback\RulePool;
+use Magento\Framework\View\Design\FileResolution\Fallback\ResolverInterface;
+use Magento\Framework\View\Design\ThemeInterface;
+use Magento\Framework\View\DesignInterface;
+
+/**
+ * Fallback resolver for design configurations.
+ *
+ * {@inheritdoc}
+ * @SuppressWarnings(PHPMD.CouplingBetweenObjects)
```

Comment:
I suggest refactoring this class to avoid introduction of new `SuppressWarnings`. 
It looks like it is coming from the old class, but refactoring does not seem to be complete as long as basic design principles are violated.

Reply:
It's mostly coming from constants and there are no clear places which can be refactored.

Reply:
Both `get` and `getParents` methods seem to have quite complicated logic with multiple nested conditions. To my understanding those may be extracted together with their specific dependencies. This would make refactoring more complete and keep the static check in place to ensure that class would not have too much responsibilities in future.